src/daemon/client.c

```diff
@@ -293,7 +293,18 @@ client_handle_set_port(struct lldpd *cfg, enum hmsg_type *type,
 	TAILQ_FOREACH(hardware, &cfg->g_hardware, h_entries)
 	    if (!strcmp(hardware->h_ifname, set->ifname)) {
 		    struct lldpd_port *port = &hardware->h_lport;
-		    (void)port;
+		    if (set->p_id && strlen(set->p_id) > 0) {
+			    log_debug("rpc", "requested change to Port ID");
+			    if (port->p_id) {
+				    free(port->p_id);
+			    }
+			    port->p_id = strdup(set->p_id);
+			    port->p_id_len = strlen(set->p_id);
+			    port->p_id_subtype = LLDP_PORTID_SUBTYPE_LOCAL;
+		    } else {
+			    /* This is the default value */
+			    port->p_id_subtype = LLDP_PORTID_SUBTYPE_LLADDR;
```

Comment:
You should do this only if `p_id` is not `NULL` (maybe when it is empty just empty). Otherwise, we can't know if that's really the port ID that we want to change. Any other port-related change would reset the port subtype to link-layer address.